### ISSUE TITLE
Release script failed on deleting 'latest_release'

### ISSUE THREAD

**Issue by bboreham** [weaveworks/weave#2866]
When I was running the release process for 1.9.4, I got this message:

```
error: could not delete the release corresponding to tag latest_release on repo weaveworks/weave
```

I could not delete it in the UI either.  I wondered if the problem could be:
> You must delete all binary files attached to a release before you can delete the release.

from http://help.github.com/articles/editing-and-deleting-releases/#deleting-a-release
but , I still couldn't delete it after deleting all the attachments, so clearing out the contents of the previous latest_release and manually replaced them with the files from 1.9.4.

**Comment by marccarre** [comment]
What would happen if we were to change the release script to stop updating this `latest_release` tag and just use [GitHub "latest" URLs](https://help.github.com/articles/linking-to-releases/) instead? Do we have any user facing script, documentation, etc. relying on it for artifacts download?

Note that:

1. https://github.com/weaveworks/weave/releases/latest redirects to the page of the latest release's **tag**, e.g. at the time of writing, https://github.com/weaveworks/weave/releases/tag/latest_release which itself points to https://github.com/weaveworks/weave/releases/tag/v1.9.4 however,
2. the redirect doesn't seem to work for release artifacts, e.g. while https://github.com/weaveworks/weave/releases/download/latest_release/weave downloads `weave` from the `latest_release` tag, the following URLs just return a 404:
    * https://github.com/weaveworks/weave/releases/latest/download/weave
    * https://github.com/weaveworks/weave/releases/download/latest/weave

**Comment by rade** [comment]
iirc the lack of redirects for download artifacts is the reason we don't use the GH 'latest' feature.

**Comment by marccarre** [comment]
Indeed. Just confirmed by @awh on Slack: https://weaveworks.slack.com/archives/C0DNGPKLY/p1490279465910167
